opentelemetry/chart/templates/logs-collector.yaml

```diff
@@ -192,15 +223,61 @@ spec:
+              - merge_maps(attributes, ExtractGrokPatterns(body, "%{GREEDYDATA:osd.wall.type}\\:%{SPACE}%{NUMBER:osd.wall.writes:float}(K|M|G)?%{SPACE}writes,%{SPACE}%{NUMBER:osd.wall.syncs:float}(K|M|G)?%{SPACE}syncs,%{SPACE}%{NUMBER:osd.wall.writes_per_sync:float}%{SPACE}writes per sync, written\\:%{SPACE}%{NUMBER:osd.wall.written_gb:float}%{SPACE}(GB|MB|TB)?,%{SPACE}%{NUMBER:osd.written_mb_sec:float}", true),"upsert")
+              - merge_maps(attributes, ExtractGrokPatterns(body, "%{WORD:log_level}%{SPACE}%{NOTSPACE}%{SPACE}%{SPACE}%{NOTSPACE:process.name}", true),"upsert")
+              - set(attributes["config.parsed"], "ceph_osd") where attributes["osd.stats.level"] != nil
+              - set(attributes["config.parsed"], "ceph_osd") where attributes["log_level"] != nil
+
+      transform/files:
+        error_mode: ignore
+        log_statements:
+          - context: log
+            conditions:
+              - resource.attributes["log.type"] == "files"
+            statements:
+              - merge_maps(attributes, ExtractGrokPatterns(body, "%{TIMESTAMP_ISO8601:timestamp}%{SPACE}%{GREEDYDATA:log}", true), "upsert")
+              - set(attributes["config.parsed"], "files") where attributes["log_level"] != nil
+
+      transform/kvm_openvswitch:
+        error_mode: ignore
+        log_statements:
+          - context: log
+            conditions:
+              - resource.attributes["k8s.daemonset.name"] == "neutron-openvswitch-agent"
+            statements:
+              - merge_maps(attributes, ExtractGrokPatterns(body, "%{TIMESTAMP_ISO8601:logtime}%{SPACE}%{NUMBER:process.id}%{SPACE}%{WORD:log_level}%{SPACE}%{NOTSPACE:process.name}%{SPACE}\\[%{REQUEST_ID:request.id}%{SPACE}%{REQUEST_ID:request.global_id}", true, ["REQUEST_ID=([A-Za-z0-9-]+)"]), "upsert")
+              - set(attributes["config.parsed"], "kvm_openvswitch") where attributes["log_level"] != nil
+
+      transform/kvm_nova_agent:
+        error_mode: ignore
+        log_statements:
+          - context: log
+            conditions:
+              - resource.attributes["k8s.daemonset.name"] == "nova-hypervisor-agents-compute-kvm"
+            statements:
+              - merge_maps(attributes, ExtractGrokPatterns(body, "%{TIMESTAMP_ISO8601:logtime}%{SPACE}%{NUMBER:process.id}%{SPACE}%{WORD:log_level}%{SPACE}%{NOTSPACE:process.name}%{SPACE}\\[%{REQUEST_ID:request.id}%{SPACE}%{REQUEST_ID:request.global_id}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}\\]%{SPACE}", true, ["REQUEST_ID=([A-Za-z0-9-]+)"]), "upsert")
```

Comment:
What is the reason for so many pairs of `%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}...` (5 pairs exactly)?

Reply:
I have shorten the parsing.